src/python-common/ceph/rgw/rgwam_core.py

```diff
@@ -386,101 +430,125 @@ def zone_op(self):
+                                                 endpoints,
+                                                 is_master=zonegroup_is_master)
+            zonegroup = EntityKey(zg_info['name'], zg_info['id'])
+            logging.info(f'Created zonegroup name={zonegroup.name} id={zonegroup.id}')
+            return zonegroup
         except RGWAMException as e:
             raise RGWAMException('failed to create zonegroup', e)
 
-        zg_name = zg_info['name']
-        zg_id = zg_info['id']
-        logging.info('Created zonegroup %s (%s)' % (zg_name, zg_id))
-
-        zg = EntityName(zg_name)
-
+    def create_zone(self, realm, zg, zone_name, zone_is_master, access_key=None,
+                    secret=None, endpoints=None):
         try:
-            zone_info = self.zone_op().create(realm, zg, EntityName(zone_name), endpoints,
-                                              is_master=True)
+            zone_info = self.zone_op().create(realm, zg,
+                                              EntityName(zone_name),
+                                              endpoints,
+                                              is_master=zone_is_master,
+                                              access_key=access_key,
+                                              secret=secret)
+
+            zone = EntityKey(zone_info['name'], zone_info['id'])
+            logging.info(f'Created zone name={zone.name} id={zone.id}')
+            return zone
         except RGWAMException as e:
             raise RGWAMException('failed to create zone', e)
 
-        zone_name = zone_info['name']
-        zone_id = zone_info['id']
-        logging.info('Created zone %s (%s)' % (zone_name, zone_id))
-
-        zone = EntityName(zone_name)
-
+    def create_system_user(self, realm, zonegroup, zone):
         try:
-            period_info = self.period_op().update(realm, EntityName(zg_name), zone, commit=True)
-        except RGWAMCmdRunException as e:
-            raise RGWAMException('failed to update period', e)
-
-        period = RGWPeriod(period_info)
-
-        logging.info('Period: ' + period.id)
-
-        try:
-            sys_user_info = self.user_op().create(zone, zg, uid=sys_uid, uid_prefix='user-sys',
+            sys_user_info = self.user_op().create(zone,
+                                                  zonegroup,
+                                                  uid=f'sysuser-{realm.name}',
+                                                  uid_prefix='user-sys',
                                                   is_system=True)
+            sys_user = RGWUser(sys_user_info)
+            logging.info('Created system user: %s' % sys_user.uid)
+            access_key = sys_user.keys[0].access_key if sys_user and sys_user.keys else ''
+            secret_key = sys_user.keys[0].secret_key if sys_user and sys_user.keys else ''
+            sys_user.add_key(access_key, secret_key)
```

Comment:
Out of curiosity what's the point of `add_key` here?

Reply:
Good catch! I think it's useless and the previous lines as well (they are there because I tried to keep the original code logic as it's). I adjusted/fixed the code in the last commit.